openeo/internal/documentation.py

```diff
@@ -1,17 +1,23 @@
 """
 Utilities to build/automate/extend documentation
 """
+
 import collections
 import inspect
 import textwrap
 from functools import partial
-from typing import Callable, Optional, Tuple
+from typing import Callable, Optional, Tuple, TypeVar
 
 # TODO: give this a proper public API?
 _process_registry = collections.defaultdict(list)
 
 
-def openeo_process(f: Optional[Callable] = None, process_id: Optional[str] = None, mode: Optional[str] = None):
+T = TypeVar("T")
```

Comment:
I'm not that familiar with TypeVar details yet, but I think it makes sense to define it with `bound=Callable` to keep the spirit of the original typehint of openeo_process

Reply:
Good suggestion, I updated this.